packages/medusa/src/api/middlewares/validators/product-existence.ts

```diff
@@ -1,26 +1,30 @@
 import { NextFunction, Request, Response } from "express"
-import { ProductService } from "../../../services"
-import { ProductBatchSalesChannel } from "../../../types/sales-channels"
 
-export function validateProductsExist(
-  getProducts: (req) => ProductBatchSalesChannel[] | undefined
+type GetProductsRequiredParams = {
+  id: string
+}
+
+export function validateProductsExist<T extends GetProductsRequiredParams>(
```

Comment:
suggestion: `T extends GetProductsRequiredParams = GetProductsRequiredParams` so that it is not mandatory to pass the template.